src/Plugin/Filter/TokenFilter.php

```diff
@@ -89,7 +104,23 @@ public static function create(ContainerInterface $container, array $configuratio
    * {@inheritdoc}
    */
   public function process($text, $langcode) {
-    return new FilterProcessResult($this->token->replace($text), [], ['langcode' => $langcode]);
+    $data = [];
+
+    // Attempt to figure out the current context based on the current backtrace.
```

Comment:
I'm not yet familiar with the core context system, but I've never been particularly fond of the approach implemented in Token Filter and have previously offered alternatives.

I only opted for this approach to ensure that it is as close to the Drupal 7 version for the sake of a port, but I am 100% open to improving this.

Will look into that now.